File: src/main/java/net/thedragonteam/armorplus/entity/entitygolem/EntityIceGolem.java
package net.thedragonteam.armorplus.entity.entitygolem;

import net.minecraft.entity.monster.EntityIronGolem;
import net.minecraft.util.ResourceLocation;
import net.minecraft.world.World;

import javax.annotation.Nullable;

import static net.thedragonteam.armorplus.util.Utils.setRL;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
public class EntityIceGolem extends EntityIronGolem {

    private ResourceLocation loot = setRL("entities/ice_golem");

    EntityIceGolem(World worldIn) {
        super(worldIn);
    }

    @Nullable
    @Override
    protected ResourceLocation getLootTable() {
        return loot;
    }

}

File: src/main/java/net/thedragonteam/armorplus/entity/entitygolem/EntityIceGolem.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.entity.entitygolem

import net.minecraft.entity.monster.EntityIronGolem
import net.minecraft.util.ResourceLocation
import net.minecraft.world.World
import net.thedragonteam.armorplus.ArmorPlus

class EntityIceGolem(worldIn: World) : EntityIronGolem(worldIn) {

    override fun getLootTable(): ResourceLocation? {
        return LOOT
    }

    companion object {
        val LOOT = ResourceLocation(ArmorPlus.MODID, "entities/ice_golem")
    }
}
